tests/integration/unbonding.go

```diff
@@ -461,3 +461,47 @@ func (s *CCVTestSuite) TestRedelegationProviderFirst() {
 	// Check that ccv unbonding op has been deleted
 	checkCCVUnbondingOp(s, s.providerCtx(), s.consumerChain.ChainID, valsetUpdateID, false)
 }
+
+func (s *CCVTestSuite) TestTooManyLastValidators() {
+	sk := s.providerApp.GetTestStakingKeeper()
+
+	// get current staking params
+	p := sk.GetParams(s.providerCtx())
+
+	// get validators, which are all active atm
+	vals := sk.GetAllValidators(s.providerCtx())
+	s.Require().Equal(len(vals), len(sk.GetLastValidators(s.providerCtx())))
+
+	// jail a validator
+	val := vals[0]
+	consAddr, err := val.GetConsAddr()
+	s.Require().NoError(err)
+	sk.Jail(s.providerCtx(), consAddr)
+
+	// save the current number of bonded vals
+	lastVals := sk.GetLastValidators(s.providerCtx())
+
+	// pass one block to apply the validator set changes
+	// (calls ApplyAndReturnValidatorSetUpdates in the the staking module EndBlock)
+	s.providerChain.NextBlock()
+
+	// verify that the number of bonded validators is decreased by one
+	s.Require().Equal(len(lastVals)-1, len(sk.GetLastValidators(s.providerCtx())))
+
+	// update maximum validator to the equal number of bonded validators
+	p.MaxValidators = uint32(len(sk.GetLastValidators(s.providerCtx())))
+	sk.SetParams(s.providerCtx(), p)
+
+	// pass one block to apply validator set changes
+	s.providerChain.NextBlock()
+
+	// unjail validator
+	sk.Unjail(s.providerCtx(), consAddr)
```

Comment:
So, I guess validators `vals` are sorted in descending order by power and that's why `vals[0]` when unjailed enters again the active set? Might be worth to write a comment.